test/compound_and_array_test.mjs

```diff
@@ -1,51 +1,94 @@
 #!/usr/bin/env node
 
 import { strict as assert } from 'assert';
-import h5wasm from "../dist/node/hdf5_hl.js";
+import h5wasm from '../dist/node/hdf5_hl.js';
 
 async function compound_array_test() {
-
   await h5wasm.ready;
-  var f = new h5wasm.File("./test/array.h5", "r");
+  var f = new h5wasm.File('./test/array.h5', 'r');
 
   assert.deepEqual(
-    f.get("float_arr").value, 
-    new Float64Array([
-      0, 1, 2, 3,
-      4, 5, 6, 7
-    ])
+    f.get('float_arr').value,
+    new Float64Array([0, 1, 2, 3, 4, 5, 6, 7])
   );
 
-  assert.deepEqual(
-    f.get("string_arr").value, 
-    [
-      'hello', 'there',
-      'hello', 'there',
-      'hello', 'there',
-      'hello', 'there'
-    ]
-  );
+  assert.deepEqual(f.get('string_arr').value, [
+    'hello',
+    'there',
+    'hello',
+    'there',
+    'hello',
+    'there',
+    'hello',
+    'there',
+  ]);
 
-  assert.deepEqual(
-    f.get("compound").value,
-    [
-      [
-        new Float64Array([ 0, 1, 2, 3 ]),
-        [ 'hello', 'there', 'hello', 'there' ]
+  assert.deepEqual(f.get('compound').value, [
+    [new Float64Array([0, 1, 2, 3]), ['hello', 'there', 'hello', 'there']],
+    [new Float64Array([4, 5, 6, 7]), ['hello', 'there', 'hello', 'there']],
+  ]);
+
+  assert.deepEqual(f.get('compound').metadata, {
+    compound_type: {
+      members: [
+        {
+          array_type: {
+            cset: -1,
+            dims: [2, 2],
+            littleEndian: true,
+            signed: false,
+            size: 8,
+            type: 1,
+            vlen: false,
+          },
+          cset: -1,
+          littleEndian: true,
+          name: 'floaty',
+          offset: 0,
+          shape: [],
+          signed: false,
+          size: 32,
+          type: 10,
+          vlen: false,
+        },
+        {
+          array_type: {
+            cset: 0,
+            dims: [2, 2],
+            littleEndian: false,
+            signed: false,
+            size: 5,
+            type: 3,
+            vlen: false,
+          },
```

Comment:
Note that there is also an inconsistency with `ArrayTypeMetadata`: it is supposed to extend `Metadata` but there is no `shape` nor` total_size` field

Reply:
I'm thinking it might make sense to rename array_type.dims to array_type.shape, and add a total_size member.  I don't think this would cause any problems, and it would address this typing issue.  There was no particularly good reason to call them "dims" instead of "shape" in the first place, except to avoid confusion with the shape of the dataset itself - and the structure of the metadata prevents this type of confusion.